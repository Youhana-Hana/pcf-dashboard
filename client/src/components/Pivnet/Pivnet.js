import React, { Component } from 'react';

import {
  Row,
  Col,
  Card,
  CardHeader,
  CardBlock,
  CardFooter,
  CardTitle
} from "reactstrap";

import CardRow from '../CardRow/';

class Pivnet extends Component {
  render() {
    return (
      <div>
        <CardHeader>

          <h1 className="mb-0">{this.props.header}</h1>

        </CardHeader>
        <Card className="text-white bg-primary">
          <CardBlock className="card-body pb-0">
            {
              this.props.versions.map((version) => {
                let row = {
                  label: version.id,
                  value: version.latest,
                  url: version.releaseNotesUrl
                };

                return <CardRow row= {row} key={version.id}></CardRow>;
              })
            }
          </CardBlock>
        </Card>
      </div>
    )
  }
}

export default Pivnet;
