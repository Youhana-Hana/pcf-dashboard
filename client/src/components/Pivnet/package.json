{
  "name": "Pivnet",
  "version": "0.0.0",
  "private": true,
  "main": "./Pivnet.js"
}
