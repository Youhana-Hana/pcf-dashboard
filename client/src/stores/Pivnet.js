import AppDispatcher from '../dispatcher/AppDispatcher';
import Constants from '../constants/PcfPipelines.js';
import { EventEmitter } from 'events';

const CHANGE_EVENT  = 'change';

let _versions = [];

function setVersion(versions) {
    _versions = versions;
}

class PivnetStoreClass extends EventEmitter {

    emitChange() {
        this.emit(CHANGE_EVENT);
    }

    addChangeListener(callback) {
        this.on(CHANGE_EVENT, callback);
    }

    removeChangeListener(callback) {
        this.removeListener(CHANGE_EVENT, callback);
    }

    getVersions() {
        return _versions;
    }
}

const PivnetStore = new PivnetStoreClass();

PivnetStore.dispatchToken = AppDispatcher.register(action => {
    switch(action.actionType) {
    case Constants.RECIEVE_PAYLOAD:
        setVersion(action.data.pivnet.ertVersions);
        PivnetStore.emitChange();
    }
});

export default PivnetStore;
