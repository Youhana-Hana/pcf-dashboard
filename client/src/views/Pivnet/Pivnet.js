import React, { Component } from 'react';
import {
  Row,
  Col,
  Card,
  CardHeader,
  CardBlock,
} from 'reactstrap';

import ERTVersions from '../../components/Pivnet/';
import OpsManager from '../../components/OpsManager/';

class Pivnet extends Component {
  render() {
    return (
        <Row>
        <Col>
        <Card>
        <CardHeader>
        <h1 className="mb-0">Pivnet</h1>
        </CardHeader>
        <CardBlock className="card-body">
        
        <Row>
        <Col xs="12" sm="6" lg="5"><ERTVersions/></Col>
        <Col xs="12" sm="6" lg="1"/>
        <Col xs="12" sm="6" lg="5"><OpsManager/></Col>
        </Row>
        </CardBlock>
        </Card>
        </Col>
        </Row>
    )}
}

export default Pivnet;
