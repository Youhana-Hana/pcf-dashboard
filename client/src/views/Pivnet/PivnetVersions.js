import React, { Component } from 'react';
import {
  Row,
  Col,
  Card,
  CardHeader,
  CardBlock,
} from 'reactstrap';

import Pivnet from '../../components/Pivnet/';
import PivnetStore from '../../stores/Pivnet';

class PivnetVersions extends Component {
  constructor(props) {
    super(props);
    this.state =  {
      ertVersions: [],
      opsManagerVersions: []
    };

    this.onChange = this.onChange.bind(this);
  }

  componentWillMount() {
    PivnetStore.addChangeListener(this.onChange);
  }

  componentWillUnmount() {
    PivnetStore.removeChangeListener(this.onChange);
  }


  onChange() {
    this.setState({
      ertVersions: PivnetStore.getErtVersions(),
      opsManagerVersions: PivnetStore.getOpsManagerVersions()
    });
  }
  render() {
    return (
      <Row>
        <Col>
          <Card>
            <CardHeader>
              <h1 className="mb-0">Pivnet</h1>
            </CardHeader>
            <CardBlock className="card-body">

              <Row>
                <Col xs="12" sm="6" lg="5"><Pivnet versions={this.state.ertVersions} header= "ERT Versions"/></Col>
                <Col xs="12" sm="6" lg="1"/>
                <Col xs="12" sm="6" lg="5"><Pivnet versions={this.state.ertVersions} header= "OpsManager Versions"/></Col>
              </Row>
            </CardBlock>
          </Card>
        </Col>
      </Row>
    )}
}

export default PivnetVersions;
