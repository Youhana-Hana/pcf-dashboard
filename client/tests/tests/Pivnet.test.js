import React from 'react';
import renderer from 'react-test-renderer';
import Pivnet  from '../../src/components/Pivnet/';


const ertVersions= [{
  "id": "ert-1.11.*",
  "latest": "1.11.10",
  "releaseDate": "2017-08-18",
  "supportEndDate": "2018-03-31",
  "releaseNotesUrl": "https://docs.pivotal.io/pivotalcf/1-11/pcf-releaseNotes/runtime-rn.html#1.11.10"
}, {
  "id": "ert-1.10.*",
  "latest": "1.10.24",
  "releaseDate": "2017-08-18",
  "supportEndDate": "2017-12-31",
  "releaseNotesUrl": "https://docs.pivotal.io/pivotalcf/1-10/pcf-releaseNotes/runtime-rn.html#1.10.24"
}, {
  "id": "ert-1.9.*",
  "latest": "1.9.37",
  "releaseDate": "2017-08-18",
  "supportEndDate": "2017-09-30",
  "releaseNotesUrl": "https://docs.pivotal.io/pivotalcf/1-9/pcf-releaseNotes/runtime-rn.html#1.9.37"
}];

it('test to see if renders correctly', () => {
  const versions = renderer
  .create(<Pivnet versions={ertVersions} />)
  .toJSON();
  expect(versions).toMatchSnapshot();
});
