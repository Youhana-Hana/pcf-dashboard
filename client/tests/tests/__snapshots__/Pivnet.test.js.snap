// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if renders correctly 1`] = `
<div>
  <div
    className="card-header"
  >
    <h1
      className="mb-0"
    />
  </div>
  <div
    className="text-white bg-primary card"
  >
    <div
      className="card-body pb-0 card-block"
    >
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            ert-1.11.*
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            <a
              className="text-white nav-link float-right"
              href="https://docs.pivotal.io/pivotalcf/1-11/pcf-releaseNotes/runtime-rn.html#1.11.10"
              target="_blank"
            >
              1.11.10
            </a>
          </h3>
        </div>
      </div>
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            ert-1.10.*
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            <a
              className="text-white nav-link float-right"
              href="https://docs.pivotal.io/pivotalcf/1-10/pcf-releaseNotes/runtime-rn.html#1.10.24"
              target="_blank"
            >
              1.10.24
            </a>
          </h3>
        </div>
      </div>
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            ert-1.9.*
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            <a
              className="text-white nav-link float-right"
              href="https://docs.pivotal.io/pivotalcf/1-9/pcf-releaseNotes/runtime-rn.html#1.9.37"
              target="_blank"
            >
              1.9.37
            </a>
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
